### PR TITLE
create favorite restaurants and fix problems

### PR DESCRIPTION
**Pull request by harutkarapetyan**

### CODE REVIEW

File: app/api/andpoints/favorite_restaurants.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            detail={"message": f"There was an error adding favorite restaurant"
+                                    f"ERROR: {error}"})
+
+    return JSONResponse(status_code=status.HTTP_200_OK,
+                        content={"message": "Favorite restaurant successfully added"},
+                        headers=headers)
+
+
+@favorite_restaurants_router.delete("/delete_favorite_restaurant/{favorite_restaurant_d}")
```
Comment (narekrubenyan) [inline app/api/andpoints/favorite_restaurants.py]: Stex restaurant_ID petqa lini yerevi

File: app/models/models.py [hunk 2]
```diff
@@ -82,8 +84,8 @@ class Restaurant(Base):
 
 
 class FavoriteRestaurant(Base):
-    __tablename__ = "favorite_restaurant"
+    __tablename__ = "favorite_restaurants"
 
     favorite_restaurant_id = Column(Integer, nullable=False, primary_key=True)
     user_id = Column(Integer, ForeignKey("users.user_id"))
-    restaurant_id = Column(Integer, ForeignKey("foods.food_id"))
+    restaurant_id = Column(Integer, ForeignKey("restaurants.restaurant_id"), unique=True)
```
Comment (narekrubenyan) [inline app/models/models.py]: unique-n xndira talis. yerb 2 user uzuma nuyn restorany favorite srqi 500 internal server errora veradarcnum....
unique-n hani

File: app/models/models.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ class FavoriteFood(Base):
 
     favorite_food_id = Column(Integer, nullable=False, primary_key=True)
     user_id = Column(Integer, ForeignKey("users.user_id"))
-    food_id = Column(Integer, ForeignKey("foods.food_id"))
+    food_id = Column(Integer, ForeignKey("foods.food_id"), unique=True)
```
Comment (narekrubenyan) [inline app/models/models.py]: stex el hani unique-n

File: app/api/andpoints/favorite_restaurants.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+favorite_restaurants_router = APIRouter(tags=["favorite_restaurants"], prefix="/favorite_restaurants")
+
+headers = {"Access-Control-Allow-Origin": "*",
+           "Access-Control-Allow-Methods": "GET, POST, PUT, DELETE, OPTIONS",
+           "Access-Control-Allow-Headers": "Content-Type, Authorization",
+           "Access-Control-Allow-Credentials": "true"}
+
+
+@favorite_restaurants_router.post("/add_favorite_restaurants")
+def add_favorite_restaurants(user_id: int, restaurant_id: int):
```
Comment (narekrubenyan) [inline app/api/andpoints/favorite_restaurants.py:16]: stugum avelacru vor nuyn user-y nuyn restorany 2rd angam chkarana favorit avelacni....
nuyny food-i hamar

**Comment by harutkarapetyan** [comment]
unique drel ei vor nuyn mardy nuyn bany chkaroxana 2 rd angam avelacni favorite

**Pull request merged**